src/bp/ml/crf-js/index.ts

```diff
@@ -0,0 +1,77 @@
+import * as sdk from 'botpress/sdk'
+import nanoid from 'nanoid'
+
+import addon, { Tagger as AddonTagger, Trainer as AddonTrainer, Options } from './addon'
+
+export class Trainer implements Omit<AddonTrainer, 'train'>, sdk.MLToolkit.CRF.Trainer {
+  private crfsuite_addon: AddonTrainer
+
+  constructor() {
+    this.crfsuite_addon = new addon.Trainer({ debug: true })
+  }
+
+  append(xseq: string[][], yseq: string[]): void {
+    return this.crfsuite_addon.append(xseq, yseq)
+  }
+
+  train_async(model_filename: string, cb: (s: string) => void): Promise<number> {
+    return this.crfsuite_addon.train_async(model_filename, cb)
+  }
+
+  get_params(options: Options) {
+    return this.crfsuite_addon.get_params(options)
+  }
+
+  set_params(options: Options): void {
+    return this.crfsuite_addon.set_params(options)
+  }
+
+  train(elements: sdk.MLToolkit.CRF.DataPoint[], params: sdk.MLToolkit.CRF.TrainerOptions): Promise<string> {
```

Comment:
this is the code with which we override the train function in the webworker so it can send a "train" message to ML_WORKER I don't think this should reside in crf-js ...

Reply:
No, you're 100% right... 

It's just the least hackish way I found to make it work and keep the same behavior it had before...

I honnestly don't like this solution, but its the best I found that still works... I'll sleep on this and think about it.